aten/src/ATen/native/native_functions.yaml

```diff
@@ -942,6 +942,19 @@
 
 - func: flatten(Tensor self, int start_dim=0, int end_dim=-1) -> Tensor
   variants: function, method
+  named_guard: False
+
+- func: flatten(Tensor self, int start_dim, int end_dim, Dimname out_dim) -> Tensor
```

Comment:
Would it make sense for `out_dim`s to be optional, defaulting to wildcard?

Would a mixed interface also be useful, something like `Dimname start_dim, int end_dim=-1`

Reply:
> Would it make sense for out_dims to be optional, defaulting to wildcard?

I think this is too much magic and the API doesn't read well. For example, for Tensor[N, C, H, W], `tensor.flatten('C', out_dim='features')` doesn't give the reader too much useful information.

> Would a mixed interface also be useful

Mixed interface is annoying to support. It's useful if the user has unnamed dimensions and have trouble referring to them, but then the workaround is to just use the `flatten(int, int, Dimname)` overload. We don't support mixed anywhere else (for example, torch.sum(...) takes a DimnameList or an IntList, but not mixed.)

Reply:
partly I ask because `flatten(Tensor self, int start_dim=0, int end_dim=-1)` exists, so seems natural to have a named equivalent. Is it necessary to always require the out_dim to be named or could it default to wildcard?

Reply:
I'm going to require the `out_dim` to be named for now. If someone is using named tensors, it's better to steer them to having more named dimensions than fewer.